samples/net/mqtt_publisher/src/main.c

```diff
@@ -294,111 +325,86 @@ static int try_to_connect(struct mqtt_client_ctx *client_ctx)
-
-	client_ctx.mqtt_ctx.peer_addr_str = SERVER_ADDR;
-	client_ctx.mqtt_ctx.peer_port = SERVER_PORT;
-
-#if defined(CONFIG_MQTT_LEGACY_LIB_TLS)
-	/** TLS setup */
-	client_ctx.mqtt_ctx.request_buf = tls_request_buf;
-	client_ctx.mqtt_ctx.request_buf_len = TLS_REQUEST_BUF_SIZE;
-	client_ctx.mqtt_ctx.personalization_data = TLS_PRIVATE_DATA;
-	client_ctx.mqtt_ctx.personalization_data_len = strlen(TLS_PRIVATE_DATA);
-	client_ctx.mqtt_ctx.cert_host = TLS_SNI_HOSTNAME;
-	client_ctx.mqtt_ctx.tls_mem_pool = &tls_mem_pool;
-	client_ctx.mqtt_ctx.tls_stack = tls_stack;
-	client_ctx.mqtt_ctx.tls_stack_size = K_THREAD_STACK_SIZEOF(tls_stack);
-	client_ctx.mqtt_ctx.cert_cb = setup_cert;
-	client_ctx.mqtt_ctx.entropy_src_cb = NULL;
+#if defined(CONFIG_MQTT_LIB_TLS)
+	rc = tls_init();
+	PRINT_RESULT("tls_init", rc);
+	if (rc < 0) {
+		return;
+	}
 #endif
 
-	/* Publisher apps TX the MQTT PUBLISH msg */
-	client_ctx.mqtt_ctx.publish_tx = publish_cb;
-
-	/* The connect message will be sent to the MQTT server (broker).
-	 * If clean_session here is 0, the mqtt_ctx clean_session variable
-	 * will be set to 0 also. Please don't do that, set always to 1.
-	 * Clean session = 0 is not yet supported.
-	 */
-	client_ctx.connect_msg.client_id = MQTT_CLIENTID;
-	client_ctx.connect_msg.client_id_len = strlen(MQTT_CLIENTID);
-	client_ctx.connect_msg.clean_session = 1;
-
-	client_ctx.connect_data = "CONNECTED";
-	client_ctx.disconnect_data = "DISCONNECTED";
-	client_ctx.publish_data = "PUBLISH";
-
 	rc = network_setup();
 	PRINT_RESULT("network_setup", rc);
 	if (rc < 0) {
 		return;
 	}
 
-	rc = mqtt_init(&client_ctx.mqtt_ctx, MQTT_APP_PUBLISHER);
+	rc = mqtt_init();
 	PRINT_RESULT("mqtt_init", rc);
 	if (rc != 0) {
 		return;
 	}
 
-	for (i = 0; i < CONN_TRIES; i++) {
-		rc = mqtt_connect(&client_ctx.mqtt_ctx);
-		PRINT_RESULT("mqtt_connect", rc);
-		if (!rc) {
-			goto connected;
-		}
-	}
-
-	goto exit_app;
-
-connected:
-
 	rc = try_to_connect(&client_ctx);
 	PRINT_RESULT("try_to_connect", rc);
 	if (rc != 0) {
-		goto exit_app;
+		return;
 	}
 
 	i = 0;
-	while (i++ < APP_MAX_ITERATIONS) {
-		rc = mqtt_tx_pingreq(&client_ctx.mqtt_ctx);
-		k_sleep(APP_SLEEP_MSECS);
-		PRINT_RESULT("mqtt_tx_pingreq", rc);
-
-		prepare_mqtt_publish_msg(&client_ctx.pub_msg, MQTT_QoS0);
-		rc = mqtt_tx_publish(&client_ctx.mqtt_ctx, &client_ctx.pub_msg);
-		k_sleep(APP_SLEEP_MSECS);
-		PRINT_RESULT("mqtt_tx_publish", rc);
-
-		prepare_mqtt_publish_msg(&client_ctx.pub_msg, MQTT_QoS1);
-		rc = mqtt_tx_publish(&client_ctx.mqtt_ctx, &client_ctx.pub_msg);
-		k_sleep(APP_SLEEP_MSECS);
-		PRINT_RESULT("mqtt_tx_publish", rc);
-
-		prepare_mqtt_publish_msg(&client_ctx.pub_msg, MQTT_QoS2);
-		rc = mqtt_tx_publish(&client_ctx.mqtt_ctx, &client_ctx.pub_msg);
-		k_sleep(APP_SLEEP_MSECS);
-		PRINT_RESULT("mqtt_tx_publish", rc);
-	}
+	while (i++ < 4 * APP_MAX_ITERATIONS && connected) {
+		iter_start = k_uptime_get();
+
+		if (qos > MQTT_QoS_2_EACTLY_ONCE) {
```

Comment:
Lower case "o" is typically found in Quality of Service acronym, I guess that's the reason it's here. It might be upper case though, for consistency.

Reply:
> It might be upper case though, for consistency.

+1 for consistency. If everything is upper-case, then CamelCase acronyms should be upper-case too.